### DISCUSSION TITLE
请问如何将waline管理系统的评论链接更改？

### DISCUSSION THREAD

**Discussion by Niko-Karen** [walinejs/waline#1160] [Q&A] [answered]
本人是使用shoka主题的hexo博客，今天从valine系统迁移到waline，用的leancloud国际版后台，但我发现了一个问题：waline的js自动返回的是这种数据。
![image](https://user-images.githubusercontent.com/91728467/176357308-430dd213-15ed-4ec6-aafc-9a2d22b28bc2.png)
但只要更改一下，去掉一个斜杠，数据就能恢复
![image](https://user-images.githubusercontent.com/91728467/176357489-182520e3-0077-4dbe-8f32-eef398aafab8.png)
但是，在waline和leancloud的后台，我并没有看到更改这种链接的地方。所以，我想请教一下大佬们。

**Comment by lizheming** [comment, marked as answer]
https://waline.js.org/reference/component.html#path

**Reply by Niko-Karen** [reply]
![image](https://user-images.githubusercontent.com/91728467/176569596-b627ca3f-a3e5-4696-8b5b-a79be87b93a0.png)
![image](https://user-images.githubusercontent.com/91728467/176569670-f30fbbb1-a60c-42e8-9c77-9ee00a5e9f06.png)

我直接套上去，好像不行……

**Reply by AnzhiZhang** [reply]
直接写引号会变成这样
![image](https://user-images.githubusercontent.com/37402126/177071912-ddbafe72-c1f8-4d84-835d-eafd329b8432.png)
可以使用 `String()` 来表示空字符串，如需要文本可以使用 `String.fromCharCode())`，例如这个消除 URL 第一段：

```javascript
window.location.pathname.replace(/^\/.*?\//, String.fromCharCode(47))
```

**Reply by Niko-Karen** [reply]
> 直接写引号会变成这样 ![image](https://user-images.githubusercontent.com/37402126/177071912-ddbafe72-c1f8-4d84-835d-eafd329b8432.png) 可以使用 `String()` 来表示空字符串，如需要文本可以使用 `String.fromCharCode())`，例如这个消除 URL 第一段：
> 
> ```js
> window.location.pathname.replace(/^\/.*?\//, String.fromCharCode(47))
> ```
![image](https://user-images.githubusercontent.com/91728467/177167475-78c6bd74-d5f7-4c60-894c-b729fa990862.png)
但是好像不行啊，我想要的是这种
![image](https://user-images.githubusercontent.com/91728467/177167588-254ff582-8da4-4443-9b2b-18898c579afb.png)

**Reply by Niko-Karen** [reply]
还有一个问题：waline使用的libravatar如何更改成cravatar

**Reply by AnzhiZhang** [reply]
只移除第一个 `/` 的话，正则需要使用 `^\/`

**Reply by Niko-Karen** [reply]
> 只移除第一个 `/` 的话，正则需要使用 `^\/`

![image](https://user-images.githubusercontent.com/91728467/177176023-a63d3b92-e255-487a-b273-900b9c421ff7.png)
![image](https://user-images.githubusercontent.com/91728467/177176124-dc15f08d-3528-4759-a789-92aa51fa558a.png)
这……

**Reply by AnzhiZhang** [reply]
我发的是正则内容啊。。。你不加框和不加引号一样
```javascript
window.location.pathname.replace(/^\//, String.fromCharCode(47))
```

**Reply by Niko-Karen** [reply]
😅其实不用正则，可以这样
![image](https://user-images.githubusercontent.com/91728467/177229056-991ee1ae-1846-49c7-8730-98c173c5d227.png)
```path: window.location.pathname.replace('/', '')```
已解决：[https://www.nikoblog.top](https://www.nikoblog.top)
